### ISSUE TITLE
[request] mark bookmarks read

### ISSUE THREAD

**Issue by microcreators** [go-shiori/shiori#247]
... and move them out of sight. archive them, hide them, etc. but keep them available and searchable in case I need to come back to them one day. Just don't show them in a default view.

**Comment by michaelx** [comment]
This would be a really helpful feature, especially when shiori is used as a read-it-later client.

**Comment by thejdev** [comment]
I agree that this would be a good feature.

As an interim workaround, I use this url as my default start page:
https://shiori-server/**?search=-tag%3A*#home**

The link above lists bookmarks with no tag (i.e. unread). Once I have read an article, I give it a tag (e.g. Read), and it disappears from the list.

**Comment by 8bitgentleman** [comment]
@thejdev I like that idea a lot. To remix it if you set this URL is your home it filters out the tag #read and shows all others

[http://YourShioriServer/?search=-read#home](http://YourShioriServer/?search=-read#home)

**Comment by deanishe** [comment]
Hi guys. I'm your new Shiori maintainer.

This feature is a high priority for me. Specifically, I want to add support for read, starred and—possibly—archived status.

Right now, I'm thinking of something along the lines that @thejdev suggests: basically just using tags, but treating certain tags (e.g. `unread`, `starred`) specially in the UI.

There's some "infrastructural" stuff that needs doing first, though.

**Comment by microcreators** [comment]
First of all, thanks for reviving the project, appreciate it. If I can suggest something here, start easy with the change. Maybe just one special tag for starters like `read`? I think I'd be simpler to add this vs. having everything else tagged as `unread`.

**Comment by deanishe** [comment]
> I think I'd be simpler to add this vs. having everything else tagged as unread.

What makes you say it would be simpler?

I think the main difference is whether you think a "read later" list should be opt-in (add `unread`) or opt-out (add `read`).

I'm more inclined to the former (tag as `unread` to add to "read later" list) because I don't want an update to automatically shunt everybody's entire list of existing bookmarks (and any new ones they add/import) into their new "read later" list (because they aren't tagged as `read`).

Existing users have probably read most of the articles they're interested in by this point, and they may not even want to use the feature. That's a better fit with the opt-in `unread` model.

**Comment by microcreators** [comment]
Honestly - it doesn't matter. 99% of my bookmarks are read and stored "just in case" (it's a knowledge base for me). I think it's a matter of default tagging. If everything is `unread` by default, there will be (a lot of) manual work to hide what's no longer fresh.

**Comment by deanishe** [comment]
> If everything is unread by default, there will be (a lot of) manual work to hide what's no longer fresh.

Exactly. That's why I prefer "tag `unread` to add to read-later list".

Of course, it's simple enough to automatically add any tag to items on database migration or when importing/adding.

So, it's really a question of opt-in vs opt-out, and currently I'm quite strongly inclined towards opt-in.
